File: Example/ObjC/ExampleViewModel.m
//
//  Created by Stefan Nebel on 28.06.20.
//  Copyright © 2020 Best-iOS-Apps. All rights reserved.
//

#import "ExampleViewModel.h"
@import BIASystemKit;
@import UIKit;

@implementation ExampleViewModel

- (nonnull instancetype)init {
  self = [super init];
  if (self) {

    BIASystemInfo *info = UIDevice.currentDevice.info;

    NSMutableArray *arr = [[NSMutableArray alloc] init];
    [arr addObject:[[NSString alloc] initWithFormat:@"You are using a '%@' (%@)",
                     info.system.formatted.name,
                     info.system.name]];
    [arr addObject:[[NSString alloc] initWithFormat:@"Your device %@ jailbroken.",
                     info.secure.isJailBroken ? @"is" : @"is not"]];
    [arr addObject:[[NSString alloc] initWithFormat:@"Your device has %ld activ of %ld processors.",
                     info.processor.activeCount,
                     info.processor.count]];
    [arr addObject:[[NSString alloc] initWithFormat:@"Your system is running since %.1f s (%@).",
                     info.system.upTime,
                     [info.system.formatted upTimeWithUnitsStyle:NSDateComponentsFormatterUnitsStyleShort]]];
    self.deviceInfos = arr;
  }
  return self;
}
@end

File: Example/Swift/ExampleViewModel.swift
//
//  Created by Stefan Nebel on 14.06.20.
//  Copyright © 2020 Best-iOS-Apps. All rights reserved.
//

import Foundation
import UIKit

extension ExampleViewController {
  final class ViewModel {

    let deviceInfos: [String] = [
      "You are using a '\(UIDevice.current.info.system.formatted.name ?? "unknown")' (\(UIDevice.current.info.system.name))",
      "Your device \(UIDevice.current.info.secure.isJailBroken ? "is" : "is not") jailbroken.",
      "Your device has \(UIDevice.current.info.processor.activeCount) activ of \(UIDevice.current.info.processor.count) processors.",
      "Your system is running since \(UIDevice.current.info.system.upTime.rounded()) ms (\(UIDevice.current.info.system.formatted.upTime(unitsStyle: .short) ?? "n.v."))."
    ]
  }
}
